### PR TITLE
Create a common module for shared rackspace monitoring functionality

### PR DESCRIPTION
**Pull request by eljrax**
A stab at removing the duplicated code from core_plugins/raxmon.py and core_plugins/raxmon_autoscale.py and moving it to a shared module.

### CODE REVIEW

**Comment by jon-walton** [comment]
i'll try find some time to go through this in the next few weeks

**Comment by eljrax** [comment]
@jon-walton bump. If you're still strapped for time, I can have someone here review it and I can merge myself?

File: raxas/monitoring.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    """ Returns a list of the cloud monitoring entities
+        belonging to the active servers in a scaling group
+    """
+    cm = pyrax.cloud_monitoring
+    active_servers = scaling_group.active_servers
+    return [entity for entity in cm.list_entities()
+            if entity.agent_id in active_servers]
+
+
+def add_entity_checks(entities, check_type, metric_name, check_config={},
```
Comment (jon-walton) [inline raxas/monitoring.py]: I'm not a fan of the mutable default argument `check_config={}` can pyrax handle a `None` for `check_type` in `create_check()` ? if not, maybe give pyrax `check_type=check_type or {}`

**Comment by jon-walton** [comment]
@eljrax sorry I completely forgot. looks fine apart from that mutable default arg

**Comment by eljrax** [comment]
Fair comment! This is only ever called once, and both modules using it sets the default, so this never popped up. 

Changed to None, since pyrax handles that fine.

**Pull request merged**